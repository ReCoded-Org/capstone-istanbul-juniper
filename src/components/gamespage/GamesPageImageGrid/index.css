.gamePageImageGrid__col__link__card {
  border-width: 2px;
  transition: all, 0.5s;
}

.gamePageImageGrid__col__link__card:hover {
  transform: scale(1.05);
  transition: all, 0.5s;
}

.gamePageImageGrid__col__link {
  width: 70%;
  margin: auto;
}

.gamePageImageGrid__col__link__card__title {
  text-align: center;
  font-size: clamp(0.9rem, 2.7vw, 3rem);
  font-family: "Salsa", cursive;
}
.gamePageImageGrid {
  background: #2a8c79;
}

.gamePageImageGrid__col {
  margin-top: 60px;
  margin-bottom: 60px;
}

.gamePageImageGrid__title {
  padding-top: 4%;
  color: white;
  font-size: max(1.7rem, 5vw);
  font-family: "Salsa", cursive;
  text-align: center;
}

.gamePageImageGrid__col__link__card__img {
  width: 100%;
}
