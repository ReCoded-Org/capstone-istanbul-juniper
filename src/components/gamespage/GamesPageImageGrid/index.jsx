import React from "react";
import { Row, Col, Card } from "antd";
import { useTranslation } from "react-i18next";
import "./index.css";
import { Link } from "react-router-dom";

const GamesPageImageGrid = ({
  // an object. Check containers/GamesPage/index.jsx to see details
  gameData,
}) => {
  const [t] = useTranslation();
  const { Meta } = Card;
  const gamePosters = Object.values(gameData).map((gameObj) => {
    return (
      <Col
        key={gameObj.gameTitle}
        xs={20}
        md={{ span: 12 }}
        className="gamePageImageGrid__col"
      >
        <div className="gamePageImageGrid__col__link">
          <Link to={`games/${gameObj.route}`}>
            <Card
              hoverable
              className="gamePageImageGrid__col__link__card"
              cover={
                <img
                  alt={gameObj.gameTitle}
                  src={gameObj.imgSrc}
                  className="gamePageImageGrid__col__link__card__img"
                />
              }
            >
              <Meta
                title={
                  <h3 className="gamePageImageGrid__col__link__card__title">
                    {gameObj.gameTitle}
                  </h3>
                }
              />
            </Card>
          </Link>
        </div>
      </Col>
    );
  });
  return (
    <div className="gamePageImageGrid">
      <h1 className="gamePageImageGrid__title">{t("gamesPage.title")}</h1>
      <Row>{gamePosters}</Row>
    </div>
  );
};

export default GamesPageImageGrid;
