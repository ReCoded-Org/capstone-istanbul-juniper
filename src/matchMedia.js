/*
 * This solves matchMedia error.
 * Should be deleted when source of error found and fixed.
 * https://jestjs.io/docs/en/manual-mocks#mocking-methods-which-are-not-implemented-in-jsdom
 */

window.matchMedia = jest.fn().mockImplementation((query) => ({
  matches: false,
  media: query,
  onchange: null,
  addListener: jest.fn(), // deprecated
  removeListener: jest.fn(), // deprecated
  addEventListener: jest.fn(),
  removeEventListener: jest.fn(),
  dispatchEvent: jest.fn(),
}));
