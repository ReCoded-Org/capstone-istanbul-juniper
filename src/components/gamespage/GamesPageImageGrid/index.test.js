import React from "react";
import "../../../matchMedia";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import GamesPageImageGrid from ".";

const MOCK_GAME_DATA = {
  testGame: {
    imgSrc: "testUrl",
    route: "testRoute",
    gameTitle: "testTitle",
  },
};

it("GamesPageImageGrid renders correctly", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <GamesPageImageGrid gameData={MOCK_GAME_DATA} />
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
