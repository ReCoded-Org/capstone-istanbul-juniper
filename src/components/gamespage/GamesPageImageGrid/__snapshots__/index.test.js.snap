// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GamesPageImageGrid renders correctly 1`] = `
<div
  className="gamePageImageGrid"
>
  <h1
    className="gamePageImageGrid__title"
  >
    gamesPage.title
  </h1>
  <div
    className="ant-row"
    style={Object {}}
  >
    <div
      className="ant-col gamePageImageGrid__col ant-col-xs-24 ant-col-md-8"
      style={Object {}}
    >
      <div
        className="gamePageImageGrid__col__link"
      >
        <a
          href="/games/testRoute"
          onClick={[Function]}
        >
          <div
            className="ant-card ant-card-bordered ant-card-hoverable gamePageImageGrid__col__link__card"
          >
            <div
              className="ant-card-cover"
            >
              <img
                alt="testTitle"
                className="gamePageImageGrid__col__link__card__img"
                src="testUrl"
              />
            </div>
            <div
              className="ant-card-body"
              style={Object {}}
            >
              <div
                className="ant-card-meta"
              >
                <div
                  className="ant-card-meta-detail"
                >
                  <div
                    className="ant-card-meta-title"
                  >
                    <h3
                      className="gamePageImageGrid__col__link__card__title"
                    >
                      testTitle
                    </h3>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;
